core/src/test/java/io/questdb/cutlass/pgwire/PGJobContextTest.java

```diff
@@ -2675,6 +2675,72 @@ public void testInsertFomNodeJsWith2Parameters_WithTableCreation() throws Except
         );
     }
 
+    @Test
+    public void testInsertBooleans() throws Exception {
+        assertMemoryLeak(() -> {
+            try (
+                    PGWireServer ignored = createPGServer(4);
+                    Connection conn = getConnection(true, true)
+            ) {
+                conn.prepareStatement(
+                        "create table booleans (value boolean, ts timestamp) timestamp(ts)"
+                ).execute();
+
+                Rnd rand = new Rnd();
+                String [] values = {"TrUE", null, "", "false", "true", "banana", "22"};
+
+                try (PreparedStatement insert = conn.prepareStatement("insert into booleans values (cast(? as boolean), ?)")) {
+                    long micros = TimestampFormatUtils.parseTimestamp("2022-04-19T18:50:00.998666Z");
```

Comment:
What's the purpose of randomizing value cast as boolean ?

Reply:
to provide an assortment of string candidates to become a boolean value without having to roll out the code into several inserts.

Reply:
Let me rephrase the question . What does iteration + random give versus simple iteration over values array ? 
Are you trying to trigger some condition in pg wire "state machine" ?

Reply:
?? nope, it is just my style, what is the smell of clouds?